Curb the warning caused by using the deprecated API

See the info on [travis-ci](https://travis-ci.org/RobotWebTools/rclnodejs/builds/379566609#L8843)

``` bash
make: Entering directory '/root/rclnodejs/build'
  CXX(target) Release/obj.target/rclnodejs/src/addon.o
  CXX(target) Release/obj.target/rclnodejs/src/executor.o
  CXX(target) Release/obj.target/rclnodejs/src/handle_manager.o
  CXX(target) Release/obj.target/rclnodejs/src/rcl_bindings.o
  CXX(target) Release/obj.target/rclnodejs/src/rcl_handle.o
  CXX(target) Release/obj.target/rclnodejs/src/rcl_utilities.o
  CXX(target) Release/obj.target/rclnodejs/src/shadow_node.o
../src/shadow_node.cpp: In member function ‘virtual void rclnodejs::ShadowNode::Execute()’:
../src/shadow_node.cpp:110:69: warning: ‘v8::Local<v8::Value> Nan::MakeCallback(v8::Local<v8::Object>, const char*, int, v8::Local<v8::Value>*)’ is deprecated [-Wdeprecated-declarations]
   Nan::MakeCallback(Nan::New(this->persistent()), "execute", 0, argv);
                                                                     ^
In file included from ../src/shadow_node.hpp:18:0,
                 from ../src/shadow_node.cpp:15:
../node_modules/nan/nan.h:971:46: note: declared here
   NAN_DEPRECATED inline v8::Local<v8::Value> MakeCallback(
                                              ^
  SOLINK_MODULE(target) Release/obj.target/rclnodejs.node
  COPY Release/rclnodejs.node
```

--- a/src/shadow_node.cpp
+++ b/src/shadow_node.cpp
@@ -107,7 +107,8 @@ NAN_METHOD(ShadowNode::SyncHandles) {
 void ShadowNode::Execute() {
   Nan::HandleScope scope;
   v8::Local<v8::Value> argv[1];
-  Nan::MakeCallback(Nan::New(this->persistent()), "execute", 0, argv);
+  Nan::AsyncResource res("shadow_node");
+  res.runInAsyncScope(Nan::New(this->persistent()), "execute", 0, argv);
 }
 
 void ShadowNode::CatchException(std::exception_ptr e_ptr) {

